### PR TITLE
New FixedDecimal

### PR DESCRIPTION
**Pull request by SeungheonOh**
We should deprecate `P.E.Fixed` in near future.

### CODE REVIEW

File: src/Plutarch/Extra/Fixed.hs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import Plutarch.Extra.Function (pflip)
+import Plutarch.Num (PNum (pfromInteger, (#*)))
+import qualified Plutarch.Numeric.Additive as A (
+  AdditiveMonoid (zero),
+  AdditiveSemigroup ((+)),
+ )
+import Plutarch.TryFrom (PTryFrom (PTryFromExcess, ptryFrom'))
+import Plutarch.Unsafe (punsafeCoerce)
+
+{- | Fixed width decimal. Decimal point will be given through typelit.
```
Comment (cjay) [inline src/Plutarch/Extra/Fixed.hs:29]: Denominator instead of decimal point.

File: src/Plutarch/Extra/Fixed.hs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+-- | @since 3.11.0
+instance KnownNat u => A.AdditiveMonoid (Term s (PFixed u)) where
+  zero = pcon . PFixed $ pconstant 0
+
+{- | Convert given decimal into Ada value. Input should be Ada value with decimals; outputs
+ will be lovelace values in integer.
+
+ @since 3.9.0
+-}
+decimalToAdaValue ::
```
Comment (cjay) [inline src/Plutarch/Extra/Fixed.hs:100]: Should be renamed.

File: src/Plutarch/Extra/Fixed.hs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+-- | @since 3.11.0
+instance KnownNat u => A.AdditiveSemigroup (Term s (PFixed u)) where
+  (+) = (+)
+
+-- | @since 3.11.0
+instance KnownNat u => A.AdditiveMonoid (Term s (PFixed u)) where
+  zero = pcon . PFixed $ pconstant 0
+
+{- | Convert given decimal into Ada value. Input should be Ada value with decimals; outputs
```
Comment (cjay) [inline src/Plutarch/Extra/Fixed.hs:95]: "fixed" instead of "decimal".

**Review by cjay — APPROVED** [review]
Looks good.

**Review by kozross — CHANGES_REQUESTED** [review]

**Comment by SeungheonOh** [comment]
Thanks for the review! `Plutarch.Extra.FixedDecimal` was the newly added module and `Plutarch.Extra.Fixed` is the old one that was remained since it's not really `decimal`. I haven't got any comment of the `FixedDecimal` so I am just making sure it's getting review as well. Thanks.

**Review by kozross — APPROVED** [review]

**Comment by cjay** [comment]
Hmm I think `(#*)` in `PFixed` should also use `pto` instead of `punsafeCoerce`, like `PFixedDecimal` does. @kozross do you know if a bit-shift will be cheaper than a division on big integers in the Plutus cost model? If yes, maybe we should add a `FixedBinary`, for power-of-2 denominators, so we can get a free performance boost in the future.

**Comment by cjay** [comment]
And should `PFixedDecimal` really have a `PIntegral` instance? It looks like it will behave correctly, but it's unexpected :thinking:

**Comment by cjay** [comment]
I just found that there is a `Data.Fixed` in base, never heard of that before. We sholud try to be compatible with the way it specifies denominators, to be able to convert to/from Haskell values. It leaves the base open, but defines stuff like `E6` which will result in having a denominator of one million. Hmm I hope Purescript has something comparable, for offchain stuff.

**Comment by cjay** [comment]
Though this would be incompatible with binary fractions to allow for bit shifting. So maybe we shouldn't do this. Hmm.

**Comment by SeungheonOh** [comment]
`Data.Fixed` in the base is in base-10, so technically it should be represented with `Plutarch.Extra.PFixedDecimal` (the one that we changed). I'm not sure if this is the best idea because naming is quite confusing this way

**Comment by kozross** [comment]
@cjay In theory, assuming my design and implementation was used, a shift would be faster. Haskell by default does not do shift conversion, even on constants, for `Integer`s: I established this with benchmarks, including for bitwise primops. However, the design and implementation I wrote for bitwise primops keeps a 'conversion firewall' between numbers and blobs of bits, so any such advantage would probably be lost for single operations.

**Pull request merged**